### PR TITLE
[Doc] Better `backport/branch-name` label documentation

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR enhances the `backport/branch-name` label documentation.

### Motivation

Developer experience.

### CODE REVIEW

**Review by KevinFairise2 — APPROVED** [review]

File: docs/dev/contributing.md [hunk 1]
```diff
@@ -290,7 +290,9 @@ labels that can be use:
   Python version shipped in the agent).
 - `need-change/operator`, `need-change/helm`: indicate that the configuration needs to be modified in the operator / helm chart as well.
 - `k8s/<min-version>`: indicate the lowest Kubernetes version compatible with the PR's feature.
-- `backport/<branch-name>`: Add this label to have your changes automatically backported to `<branch-name>`.
+- `backport/<branch-name>`: Add this label to automatically create a PR against the `<branch-name>` branch with your backported changes. The backport PR creation is triggered:
+  - When a PR with the label is merged
+  - When an already merged PR gets the label.
```
Comment (buraizu) [inline docs/dev/contributing.md:295]:
```suggestion
  - When an already-merged PR gets the label
```

**Review by buraizu — APPROVED** [review]
Approving with a couple of minor updates requested

**Pull request merged**